File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as s]))

(defn anagrams-for [word prospect-list]
  (let [word-lower (s/lower-case word)
        word-chars (sort word-lower)]
    (filter #(and (= word-chars (sort (s/lower-case %1)))
                  (not= word-lower (s/lower-case %1))) prospect-list)))

File: haskell/anagram/src/Anagram.hs
module Anagram (anagramsFor) where

import Data.Char
import Data.List

isAnagram :: String -> String -> Bool
isAnagram xs ys = lxs /= lys && sort lxs == sort lys
  where
    lxs = map toLower xs
    lys = map toLower ys

anagramsFor :: String -> [String] -> [String]
anagramsFor xs = filter (isAnagram xs)
